test/temporal.jl

```diff
@@ -0,0 +1,173 @@
+        # with 126 full days in the middle
+        expected = [9:23..., repeat(0:23, 126)..., 0:9...]
+
+        @test Transforms.apply(x, hod) == expected
+        @test hod(x) == expected
+
+        # Test the tranform was not mutating
+        @test x != expected
+
+        @testset "StepRange" begin
+            x = DateTime(2020, 1, 1, 9, 0):Hour(1):DateTime(2020, 5, 7, 9, 0)
+            @test Transforms.apply(x, hod) == expected
+            @test hod(x) == expected
+        end
+
+        @testset "dims = $d" for d in (Colon(), 1)
+            @test Transforms.apply(x, hod; dims=d) == expected
+            @test hod(x; dims=d) == expected
+        end
+
+        @test_throws BoundsError Transforms.apply(x, hod; dims=2)
+
+        @testset "inds" begin
+            @test Transforms.apply(x, hod; inds=2:5) == expected[2:5]
+            @test Transforms.apply(x, hod; dims=:) == expected
+            @test Transforms.apply(x, hod; dims=1) == expected
+            @test Transforms.apply(x, hod; dims=1, inds=[2, 3, 4, 5]) == expected[2:5]
+        end
+
+        @testset "DST" begin
+            x = ZonedDateTime(2020, 3, 7, 9, 0, tz"America/New_York"):Hour(1):ZonedDateTime(2020, 3, 8, 9, 0, tz"America/New_York")
+
+            # expected result skips the DST transition hour of 2
+            expected_dst = [9:23..., 0, 1, 3:9...]
+
+            @test Transforms.apply(x, hod) == expected_dst
+            @test hod(x) == expected_dst
+        end
+    end
+
+    @testset "Matrix" begin
+        x = [
+            DateTime(2020, 1, 1, 1, 0):Hour(1):DateTime(2020, 1, 1, 3, 0);
+            DateTime(2020, 1, 1, 9, 0):Hour(1):DateTime(2020, 1, 1, 11, 0)
+        ]
+        M = reshape(x, 3, 2)
+        expected = [1 9; 2 10; 3 11]
+
+        @test Transforms.apply(M, hod) == expected
+        @test hod(M) == expected
+
+        # Test the tranform was not mutating
+        @test M != expected
+
+        @testset "dims = $d" for d in (Colon(), 1, 2)
+            @test Transforms.apply(M, hod; dims=d) == expected
+            @test hod(M; dims=d) == expected
+        end
+
+        @testset "inds" begin
+            @test Transforms.apply(M, hod; inds=[2, 3]) == expected[[2, 3]]
+            @test Transforms.apply(M, hod; dims=:, inds=[2, 3]) == expected[[2, 3]]
+            @test Transforms.apply(M, hod; dims=1, inds=[2]) == reshape(expected[[2, 5]], 1, 2)
+            @test Transforms.apply(M, hod; dims=2, inds=[2]) == reshape(expected[[4, 5, 6]], 3, 1)
+        end
+    end
+
+    @testset "AxisArray" begin
+        x = [
+            DateTime(2020, 1, 1, 1, 0):Hour(1):DateTime(2020, 1, 1, 2, 0);
+            DateTime(2020, 1, 1, 9, 0):Hour(1):DateTime(2020, 1, 1, 10, 0);
+            DateTime(2020, 1, 1, 10, 0):Hour(1):DateTime(2020, 1, 1, 11, 0)
+        ]
+        M = reshape(x, 2, 3)
+        A = AxisArray(M, foo=["a", "b"], bar=["x", "y", "z"])
+        expected = [1 9 10; 2 10 11]
```

Comment:
```suggestion
        expected = [1 9 11; 2 10 12]
```

want each element to be unique so we're sure `inds` is getting the right ones